src/LanguageServer/Impl/Implementation/LineFormatter.cs

```diff
@@ -303,45 +321,38 @@ public class LineFormatter {
                         break;
 
                     case TokenKind.ExplicitLineJoin:
-                        builder.EnsureEndsWithSpace();
+                        builder.EnsureEndsWithWhiteSpace();
                         builder.Append("\\"); // Hardcoded string so that any following whitespace doesn't make it in.
                         break;
 
                     case TokenKind.BackQuote:
                         builder.Append(token);
                         break;
 
-                    default:
-                        if (token.Kind == TokenKind.KeywordLambda) {
-                            if (token.IsInsideFunctionArgs && prev?.Kind == TokenKind.Assign) {
-                                builder.Append(token);
-
-                                if (next?.Kind != TokenKind.Colon) {
-                                    builder.EnsureEndsWithSpace();
-                                }
+                    case TokenKind.KeywordLambda when token.IsInsideFunctionArgs && prev?.Kind == TokenKind.Assign:
+                        builder.Append(token);
 
-                                break;
-                            }
+                        if (next?.Kind != TokenKind.Colon) {
+                            builder.EnsureEndsWithWhiteSpace();
                         }
 
+                        break;
+
+                    default:
                         if (token.IsKeyword) {
                             if (prev != null && !prev.IsOpen) {
-                                builder.EnsureEndsWithSpace();
+                                builder.EnsureEndsWithWhiteSpace();
                             }
 
                             builder.Append(token);
 
                             if (next != null && next.Kind != TokenKind.Colon && next.Kind != TokenKind.Semicolon) {
-                                builder.EnsureEndsWithSpace();
+                                builder.EnsureEndsWithWhiteSpace();
                             }
-
-                            break;
```

Comment:
Well, there is actually less nesting right now than before the change.
Breaking early may be better or worse. For example, if for some long code path we have simple condition that can be checked at the very beginning, it is better to terminate earlier. That is usually some null checks, state checks, etc.
But, if we have several complex sets of conditions, it **may** be better to have only one exit point. That can help to avoid bugs when code is changed later. Code like this:
```cs
if (A && !B) {
    if (C || D) {
        DoX();
        return;
    }
    DoY();
    if (!C && E) {
        DoX();
        return;
    }
}

if (C || B) {
    DoY();
    if (A && E) {
        DoX();
        return;
    }
}

// some more cases like that

DoZ();
```
is very hard to understand, and even harder to modify. And having `return` doesn't help with less nesting, it actually encourages to have more nesting, because you don't care about getting out.

So there is no general rule here, it depends on the existing code and its semantics. But with the `switch/case`, it is very uncommon for .net developer to see `break` somewhere except the end of the case. While it is totally allowed by compiler, `break` in the middle of the statement is usually associated with the early loop termination. People can simply skip it during review or code changes.

Reply:
I guess in my mind, I'm seeing this case (which is I should point out is only keyword handling, because every keyword has its own kind and therefore would be irritating to properly write in cases):

```csharp
if (bar) {
    // ...
    break;
}

// New cases go here

// Last-ditch effort
// ...
break;
```

Being changed to:

```csharp
if (bar) {
    // ...
} else {
    // New cases go here

    // Last-ditch effort
    // ...
}

break;
```

Which doesn't strike me as better (coming from writing Go almost exclusively for the past 3+ years where "exit early" is strongly preferred). Honestly, I wish there were a nice way to write this something like this:

```csharp
default when token.IsKeyword:
    // Stuff inside the if
    break;

default:
    // AppendTokenEnsureWhiteSpacesAround
    break
```

And avoid this altogether, but I can't say I know enough about C#'s switch-case pattern matching to know how to do that without completely changing what is being switched on.

Reply:
Better C# pattern matching is yet to be implemented in 8.0, along with new functional syntax. With what we have now, constructions like
```cs
default when token.IsKeyword:
    break;
```
aren't possible. They only thing that can be done is nested switch:
```cs
default:
    switch (token) {
        case TokenExt _ when token.IsKeyword:
            if (prev != null && !prev.IsOpen) {
                builder.EnsureEndsWithWhiteSpace();
            }

            builder.Append(token);

            if (next != null && next.Kind != TokenKind.Colon && next.Kind != TokenKind.Semicolon) {
                builder.EnsureEndsWithWhiteSpace();
            }
            break;

        default:
            // No tokens should make it to this case, but try to keep things separated.
            AppendTokenEnsureWhiteSpacesAround(builder, token);
            break;
    }
    break;
```

but I don't think it is better.

Reply:
Right, I figured there wasn't a way, just thinking aloud about what I wish I could have written. I'll have to look at the 8.0 spec to see what's changing. (I'm used to Go switch statements which allow a few more cases.)

I'll resolve this conversation since I think that line/col numbers are now more consistent.

Reply:
Err, rather, resolve because the nesting is alright. This isn't the line/col conversation (oops).